### PR TITLE
Add the cases of the general commands supported by virt-admin

### PR DESCRIPTION
**Pull request by iccaszhulili**
This commit contains two kinds of tests: normal_test and error_test.
The general commands are: echo, cd, pwd, exit, quit.

Signed-off-by: Lily Zhu <lizhu@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import logging
+import aexpect
+
+from autotest.client import utils
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: `from avocado.utils import process`

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    pwd_extra = params.get("pwd_extra", "")
+    echo_extra = params.get("echo_extra", "")
+    exit_cmd = params.get("exit_cmd", "exit")
+    echo_option = params.get("echo_option", "")
+    echo_str = params.get("echo_str", "xyz")
+    invalid_status_error = params.get("invalid_status_error", "no")
+    cd_status_error = params.get("cd_status_error", "no")
+    pwd_status_error = params.get("pwd_status_error", "no")
+    echo_status_error = params.get("echo_status_error", "no")
+    #readonly = "yes" == params.get("readonly", "no")
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: it can be removed.

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    result = vp.command(invalid_cmd, ignore_status=True, debug=True)
+    status = result.exit_status
+    if invalid_status_error == "yes":
+        if status == 0:
+            raise error.TestFail("Run successful with wrong command!")
+        else:
+            logging.info("Run command failed as expected.")
+    else:
+        if status != 0:
+            raise error.TestFail("Run failed with right command!")
```
Comment (will-Do) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: Raise the real error

**Comment by iccaszhulili** [comment]
@balamuruhans @will-Do I have updated the PR according to your review. Please help me review it.

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    # Run virtadmin command in interactive mode
+    vp = virt_admin.VirtadminPersistent()
+
+    # Run invalid command
+    result = vp.command(invalid_cmd, ignore_status=True, debug=True)
+    status = result.exit_status
+    if invalid_status_error == "yes":
+        if status == 0:
+            raise exceptions.TestFail("Run successful with wrong command!")
+        else:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: else statement is really not required.

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        if status != 0:
+            raise exceptions.TestFail("Run failed with right command!\n%s", result)
+
+    # Run cd command
+    result = vp.cd(cd_option, cd_extra, ignore_status=True, debug=True)
+    cd_status = result.exit_status
+    if cd_status_error == "yes":
+        if cd_status == 0:
+            raise exceptions.TestFail("Run successful with wrong command!")
+        else:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: same here

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            raise exceptions.TestFail("Run failed with right command!\n%s", result)
+
+    # Run pwd command
+    result = vp.pwd(pwd_extra, ignore_status=True, debug=True)
+    status = result.exit_status
+    output = result.stdout.strip()
+    if pwd_status_error == "yes":
+        if status == 0:
+            raise exceptions.TestFail("Run successful with wrong command!")
+        else:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: same here

File: libvirt/tests/src/virt_admin/virt_admin_itself.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+    # Run echo command
+    options = "%s %s" % (echo_option, echo_extra)
+    result = vp.echo(echo_str, options, ignore_status=True, debug=True)
+    status = result.exit_status
+    output = result.stdout.strip()
+    if echo_status_error == "yes":
+        if status == 0:
+            raise exceptions.TestFail("Run successful with wrong command!")
+        else:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: here

File: libvirt/tests/src/virt_admin/virt_admin_itself.py (hunk 4 above)
Comment (will-Do) [inline libvirt/tests/src/virt_admin/virt_admin_itself.py]: It's just a coding trick, we can live with it. Let's do it better for next patch

**Pull request merged**